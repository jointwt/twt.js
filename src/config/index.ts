import { APIConfig } from '../types.d';

export default {
  prefix: 'api/v1',
  api: {
    ping: {
      endpoint: 'ping',
      method: 'GET',
    },
    register: {
      endpoint: 'register',
      method: 'POST',
    },
    auth: {
      endpoint: 'auth',
      method: 'POST',
    },
    post: {
      endpoint: 'post',
      method: 'POST',
    },
    timeline: {
      endpoint: 'timeline',
      method: 'POST',
    },
    discover: {
      endpoint: 'discover',
      method: 'POST',
    },
    follow: {
      endpoint: 'follow',
      method: 'POST',
    },
  },
} as APIConfig;
