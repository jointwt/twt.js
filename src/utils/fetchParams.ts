import config from '../config';
import { EndpointName } from '../types.d';

export default (url: string, endpointName: EndpointName) => {
  const { endpoint, method } = config.api[endpointName];
  return {
    url: `${url}/${config.prefix}/${endpoint}`,
    method,
  };
};
