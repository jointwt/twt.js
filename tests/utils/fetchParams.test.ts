import fetchParams from '../../src/utils/fetchParams';
import config from '../../src/config';
import { EndpointName } from '../../src/types.d';

const endpoints = Object.keys(config.api);
const url = 'https://sample.twt.instance';

test('should generate correct fetch params for all known endpoint names', () => {
  endpoints.forEach(endpoint => {
    expect(fetchParams(url, endpoint as EndpointName)).toMatchSnapshot();
  })
});
