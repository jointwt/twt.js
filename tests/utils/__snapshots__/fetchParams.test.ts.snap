// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate correct fetch params for all known endpoint names 1`] = `
Object {
  "method": "GET",
  "url": "https://sample.twt.instance/api/v1/ping",
}
`;

exports[`should generate correct fetch params for all known endpoint names 2`] = `
Object {
  "method": "POST",
  "url": "https://sample.twt.instance/api/v1/register",
}
`;

exports[`should generate correct fetch params for all known endpoint names 3`] = `
Object {
  "method": "POST",
  "url": "https://sample.twt.instance/api/v1/auth",
}
`;

exports[`should generate correct fetch params for all known endpoint names 4`] = `
Object {
  "method": "POST",
  "url": "https://sample.twt.instance/api/v1/post",
}
`;

exports[`should generate correct fetch params for all known endpoint names 5`] = `
Object {
  "method": "POST",
  "url": "https://sample.twt.instance/api/v1/timeline",
}
`;

exports[`should generate correct fetch params for all known endpoint names 6`] = `
Object {
  "method": "POST",
  "url": "https://sample.twt.instance/api/v1/discover",
}
`;

exports[`should generate correct fetch params for all known endpoint names 7`] = `
Object {
  "method": "POST",
  "url": "https://sample.twt.instance/api/v1/follow",
}
`;
